src/vs/editor/contrib/format/formatActions.ts

```diff
@@ -231,7 +231,10 @@ class FormatDocumentAction extends EditorAction {
 	async run(accessor: ServicesAccessor, editor: ICodeEditor): Promise<void> {
 		if (editor.hasModel()) {
 			const instaService = accessor.get(IInstantiationService);
-			await instaService.invokeFunction(formatDocumentWithSelectedProvider, editor, FormattingMode.Explicit, Progress.None, CancellationToken.None);
+			const progressService = accessor.get(IEditorProgressService);
+			await progressService.showWhile(
```

Comment:
In the editor land we call this with 250ms delay so that it isn't too nervous, for reference see https://github.com/microsoft/vscode/blob/6a9f98bc864bd879cbc6d556983ab05781942c16/src/vs/editor/contrib/rename/rename.ts#L248

Reply:
make sense~